modules/appium/src/main/java/com/codeborne/selenide/appium/SelenideAppiumPageFactory.java

```
@@ -156,4 +156,10 @@ private static class SelenideAppiumList extends SelenideAppiumCollection impleme
   private boolean isPlatformAnnotationAdded(Field field) {
     return platformAnnotations.stream().anyMatch(field::isAnnotationPresent);
   }
+
+  @SuppressWarnings("unchecked")
+  private <T extends SelenideAppiumElement> Class<T> getTargetType(Field field) {
+    return SelenideAppiumElement.class.isAssignableFrom(field.getType()) ?
+      (Class<T>) field.getType() : (Class<T>) SelenideAppiumElement.class;
+  }
```

Maybe we should throw an exception if field is not a `SelenideAppiumElement` subclass?

---

There wasn't such check before, we just got `IllegalArgumentException` when try to `Field.set()`. But, sure, I think it will be useful.